hbase-balancer/src/main/java/org/apache/hadoop/hbase/master/balancer/BalancerClusterState.java

```diff
@@ -446,6 +447,11 @@ private void registerRegion(RegionInfo region, int regionIndex, int serverIndex,
             : serversToIndex.get(loc.get(i).getAddress()));
       }
     }
+
+    int numReplicas = region.getReplicaId() + 1;
+    if (numReplicas > maxReplicas) {
+      maxReplicas = numReplicas;
+    }
```

Comment:
This seemed like a very cheap way to determine the max replica count without actually seeing the table descriptors, which is a limitation of the current balancer implementation.

Sometime I would like to fix this limitation and make the balancer table descriptor aware, but I think that is its own problem

Reply:
I'm pretty sure that you have insufficient information here. I'm looking at the call hierarchy of `registerRegion` and I believe that there's no limit by table. So you're approximating the maximum number of region replicas used by any table in the cluster? I suspect that you need to introduce a new index by table name to the number of replicas per table.

Reply:
Agreed, we could make this better with table specific replica count tracking

Reply:
Okay but that additional tracking is not a correctness issue for how this value is used?